dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java

```diff
@@ -417,6 +418,108 @@ private void readCache(boolean firstInit) {
         }
     }
 
+    /**
+     * Answer the versioned feature shortName if exists for the passed platform version or null if doesn't exist.
+     *
+     * @return String versionedFeatureName
+     */
+    public String getVersionlessFeatureVersionForPlatform(String versionlessFeatureName, String platformName) {
+        SubsystemFeatureDefinitionImpl versionlessFeature;
+        Map features = getAllFeatures();
+        versionlessFeature = (SubsystemFeatureDefinitionImpl) features.get("io.openliberty.versionless." + versionlessFeatureName);
+        if (versionlessFeature != null && versionlessFeature.isVersionless()) {
+            for (ProvisioningFeatureDefinition child : findAllPossibleVersions(versionlessFeature)) {
+                if (child.getPlatformNames().contains(platformName))
+                    return child.getIbmShortName();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Answer the list of public versioned features derived from the passed versionless feature or empty List if doesn't exist.
+     *
+     * @return List<ProvisioningFeatureDefinition>
+     */
+    private List<ProvisioningFeatureDefinition> findAllPossibleVersions(SubsystemFeatureDefinitionImpl versionlessFeature) {
+        List<ProvisioningFeatureDefinition> result = new ArrayList<>();
+        for (FeatureResource dependency : versionlessFeature.getConstituents(null)) {
+            result.add(getVersionedFeature(dependency.getSymbolicName()));
+
+            String baseName = getFeatureBaseName(dependency.getSymbolicName());
+            List<String> tolerates = dependency.getTolerates();
+            if (tolerates != null) {
+                for (String toleratedVersion : tolerates) {
+                    String featureName = baseName + toleratedVersion;
+                    result.add(getVersionedFeature(featureName));
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     *
+     * Answer the public versioned feature based on the internal versionless linking feature
+     *
+     * @param versionlessLinkingFeatureName
+     * @return ProvisioningFeatureDefinition
+     */
+    private ProvisioningFeatureDefinition getVersionedFeature(String versionlessLinkingFeatureName) {
+        ProvisioningFeatureDefinition result = null;
+        ProvisioningFeatureDefinition feature = getFeature(versionlessLinkingFeatureName);
+        if (feature != null) {
+            //This is the versionless linking feature pointing to a public versioned feature
+            for (FeatureResource versionedFeature : feature.getConstituents(null)) {
+                //There should be only one element - set the result
+                result = getFeature(versionedFeature.getSymbolicName());
```

Comment:
There will be two elements for microprofile linking features. They contain both the mpVersion feature and the versioned feature. Adding the below snippet would fix that.
```
ProvisioningFeatureDefinition versionedFeatureDef = getFeature(versionedFeature.getSymbolicName());
if(versionedFeature.getVisibility() != Visibility.PUBLIC){
    continue;
}
result = versionedFeatureDef;
```

Reply:
Fixed up...